impeller/scene/animation/animation_player.cc

```diff
@@ -31,8 +36,18 @@ AnimationClip& AnimationPlayer::AddAnimation(
         {binding.node, binding.node->GetLocalTransform()});
   }
 
-  clips_.push_back(std::move(clip));
-  return clips_.back();
+  clips_.insert({animation->GetName(), std::move(clip)});
```

Comment:
`std::map`s `insert` or `insert_or_assign` returns an iterator to element already. So you don't need to insert and then in turn find the inserted element.